src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationMiddleware.cs

```diff
@@ -46,9 +45,14 @@ public class OpenIdConnectAuthenticationMiddleware : AuthenticationMiddleware<Op
         {
             _logger = loggerFactory.CreateLogger<OpenIdConnectAuthenticationMiddleware>();
 
+            if (string.IsNullOrEmpty(Options.SignInScheme))
+            {
+                Options.SignInScheme = externalOptions.Options.SignInScheme;
+            }
```

Comment:
Well if that really is important, we can, it just requires an additional argument corresponding to the named instance of the ExternalAuthenticationOptions to use, typically there will be only one, but for the multiple middleware scenario, names need to specified to be able to resolve the specific instance of the ExternalAuthenticationOptions.  But given that this scenario doesn't work today with the other social middlewares, I'm not sure how important it is for OIDC